extensions/basic_auth/plugin.cc

```diff
@@ -0,0 +1,258 @@
+                      request_path = JsonGetField<std::string>(
+                                         configuration["request_path"], pattern)
+                                         .value_or("");
+                      return true;
+                    })) {
+              LOG_WARN("Failed to parse configuration for request path.");
+              return false;
+            }
+            if (request_path == "") {
+              LOG_WARN("Path inside request_path field is empty.");
+              return false;
+            }
+            if (match != "prefix" && match != "exact" && match != "suffix") {
+              LOG_WARN(
+                  absl::StrCat("match_pattern: ", match, " is not valid."));
+              return false;
+            }
+
+            if (!JsonArrayIterate(
+                    configuration, "request_methods",
+                    [&](const json& method) -> bool {
+                      auto method_string = JsonValueAs<std::string>(method);
+                      if (method_string.second !=
+                          Wasm::Common::JsonParserResultDetail::OK) {
+                        return false;
+                      }
+                      request_methods.push_back(method_string.first.value());
+                      return true;
+                    })) {
+              LOG_WARN("Failed to parse configuration for request methods.");
+              return false;
+            }
+            struct BasicAuthConfigRule rule;
+            if (!JsonArrayIterate(
+                    configuration, "credentials",
+                    [&](const json& credentials) -> bool {
+                      auto credential = JsonValueAs<std::string>(credentials);
+                      if (credential.second !=
+                          Wasm::Common::JsonParserResultDetail::OK) {
+                        return false;
+                      }
+                      rule.encoded_credentials.insert(
+                          Base64::encode(credential.first.value().data(),
+                                         credential.first.value().size()));
+                      return true;
+                    })) {
+              LOG_WARN("Failed to parse configuration for credentials.");
+              return false;
+            }
+
+            if (match == "prefix") {
+              rule.pattern = Prefix;
+            } else if (match == "exact") {
+              rule.pattern = Exact;
+            } else if (match == "suffix") {
+              rule.pattern = Suffix;
+            }
+            rule.request_path = request_path;
+            for (auto& method : request_methods) {
+              basic_auth_configuration_[method].push_back(rule);
+            }
+            return true;
+          })) {
+    LOG_WARN(absl::StrCat("cannot parse plugin configuration JSON string: ",
+                          configuration_data->view()));
+  }
+  return true;
+}
+
+FilterHeadersStatus PluginRootContext::credentialsCheck(
```

Comment:
This does not need to be method of RootContext right? It can just be an util function in anonymous namespace.